packages/aws-cdk/test/api/bootstrap2.test.ts

```diff
@@ -0,0 +1,56 @@
+let mockDeployStackOptions: any;
+const mockDeployStack = jest.fn((options) => { mockDeployStackOptions = options; });
+
+jest.mock('../../lib/api/deploy-stack', () => ({
+  deployStack: mockDeployStack,
+}));
+
+import { bootstrapEnvironment2 } from '../../lib/api/bootstrap/bootstrap-environment2';
+import { MockSDK } from '../util/mock-sdk';
+
+describe('Bootstrapping v2', () => {
+  const env = {
+    account: '123456789012',
+    region: 'us-east-1',
+    name: 'mock',
+  };
+  const sdk = new MockSDK();
+
+  test('passes the bucket name as a CFN parameter', async () => {
+    await bootstrapEnvironment2(env, sdk, 'mockStack', undefined, {
+      bucketName: 'my-bucket-name',
+    });
+
+    expect(mockDeployStack).toHaveBeenCalled();
+
+    expect(mockDeployStackOptions).toMatchObject({
+      parameters: {
+        FileAssetsBucketName: 'my-bucket-name',
+    }});
+  });
+
+  test('passes the KMS key ID as a CFN parameter', async () => {
+    await bootstrapEnvironment2(env, sdk, 'mockStack', undefined, {
+      kmsKeyId: 'my-kms-key-id',
+    });
+
+    expect(mockDeployStack).toHaveBeenCalled();
+
+    expect(mockDeployStackOptions).toMatchObject({
```

Comment:
You could have done

```
expect(mockDeployStack).toHaveBeenCalledWith({
      parameters: {
        FileAssetsBucketKmsKeyId: 'my-kms-key-id', 
     }
});
```

So you don't have to copy the options parameter out by hand.

Reply:
Nope:

```
Error: expect(jest.fn()).toHaveBeenCalledWith(...expected)

- Expected
+ Received

  Object {
+   "execute": undefined,
    "parameters": Object {
+     "CloudFormationExecutionPolicies": undefined,
      "FileAssetsBucketKmsKeyId": "my-kms-key-id",
+     "FileAssetsBucketName": undefined,
+     "TrustedAccounts": undefined,
+   },
+   "roleArn": undefined,
+   "sdk": MockSDK {
+     "credentialsCache": CredentialsCache {
+       "cache": Object {},
+       "defaultAwsAccount": DefaultAWSAccount {
+         "accountCache": AccountAccessKeyCache {
+           "cacheFile": "/Users/adamruka/.cdk/cache/accounts.json",
+         },
+         "defaultAccountFetched": false,
+         "defaultAccountId": undefined,
+         "defaultCredentialsProvider": Promise {},
+         "region": [Function anonymous],
+       },
+       "defaultCredentialProvider": Promise {},
+     },
+     "defaultAwsAccount": DefaultAWSAccount {
+       "accountCache": AccountAccessKeyCache {
+         "cacheFile": "/Users/adamruka/.cdk/cache/accounts.json",
+       },
+       "defaultAccountFetched": false,
+       "defaultAccountId": undefined,
+       "defaultCredentialsProvider": Promise {},
+       "region": [Function anonymous],
+     },
+     "profile": undefined,
+     "retryOptions": Object {
+       "maxRetries": 6,
+       "retryDelayOptions": Object {
+         "base": 300,
+       },
+     },
+     "sandbox": Sandbox {
+       "createStubInstance": [Function createStubInstance],
+       "fake": [Function fake],
+       "getFakes": [Function getFakes],
+       "getRestorers": [Function anonymous],
+       "inject": [Function inject],
+       "mock": [Function mock],
+       "replace": [Function replace],
+       "replaceGetter": [Function replaceGetter],
+       "replaceSetter": [Function replaceSetter],
+       "reset": [Function reset],
+       "resetBehavior": [Function resetBehavior],
+       "resetHistory": [Function resetHistory],
+       "restore": [Function restore],
+       "restoreContext": [Function restoreContext],
+       "serverPrototype": Object {
+         "addRequest": [Function addRequest],
+         "configure": [Function configure],
+         "create": [Function create],
+         "getHTTPMethod": [Function getHTTPMethod],
+         "getRequest": [Function getRequest],
+         "handleRequest": [Function handleRequest],
+         "log": [Function log],
+         "logError": [Function logError],
+         "logger": [Function logger],
+         "processRequest": [Function processRequest],
+         "reset": [Function reset],
+         "resetBehavior": [Function resetBehavior],
+         "resetHistory": [Function resetHistory],
+         "respond": [Function respond],
+         "respondAll": [Function respondAll],
+         "respondWith": [Function respondWith],
+         "restore": [Function restore],
        },
+       "spy": [Function spy],
+       "stub": [Function stub],
+       "useFakeServer": [Function useFakeServer],
+       "useFakeTimers": [Function useFakeTimers],
+       "useFakeXMLHttpRequest": [Function useFakeXMLHttpRequest],
+       "usingPromise": [Function usingPromise],
+       "verify": [Function verify],
+       "verifyAndRestore": [Function verifyAndRestore],
+     },
+   },
+   "stack": CloudFormationStackArtifact {
+     "_dependencyIDs": Array [],
+     "_deps": Array [],
+     "assembly": CloudAssembly {
+       "artifacts": Array [
+         [Circular],
+       ],
+       "directory": "/var/folders/gl/mhgz0kz11lqgc0dklyry83xnyfk1zw/T/cdk-bootstrap-new7TOYIv",
+       "manifest": Object {
+         "artifacts": Object {
+           "mockStack": Object {
+             "environment": "aws://123456789012/us-east-1",
+             "properties": Object {
+               "templateFile": "mockStack.template.json",
+             },
+             "type": "aws:cloudformation:stack",
+           },
+         },
+         "version": "1.21.0",
+       },
+       "runtime": Object {
+         "libraries": Object {},
+       },
+       "version": "1.21.0",
+     },
+     "assets": Array [],
+     "displayName": "mockStack",
+     "environment": Object {
+       "account": "123456789012",
+       "name": "aws://123456789012/us-east-1",
+       "region": "us-east-1",
+     },
+     "id": "mockStack",
+     "manifest": Object {
+       "environment": "aws://123456789012/us-east-1",
+       "properties": Object {
+         "templateFile": "mockStack.template.json",
+       },
+       "type": "aws:cloudformation:stack",
+     },
+     "messages": Array [],
+     "name": "mockStack",
+     "originalName": "mockStack",
+     "parameters": Object {},
+     "stackName": "mockStack",
+     "template": {
...
+     },
+     "templateFile": "mockStack.template.json",
+   },
+   "tags": undefined,
  },

Number of calls: 1

    at Object.test (/Users/adamruka/workplace/cdk/aws-cdk/packages/aws-cdk/test/api/bootstrap2.test.ts:32:33)
    at <anonymous>
```

Reply:
Let me rephrase that. You could have done:

```ts
expect(mockDeployStack).toHaveBeenCalledWith(expect.objectContaining({
      parameters: {
        FileAssetsBucketKmsKeyId: 'my-kms-key-id', 
     }
}));
```

(Maybe it needs an additional `expect.objectContaining()` around the `parameters` object)

Reply:
Nope, the first snippet does work, and it is better than what I had. Changed.